### PR TITLE
Black and flake8 checks.

### PR DESCRIPTION
**Pull request by Luisella21**
Applied  black to the code base, run flake8 and fixed the relevant errors. Changed circleci configuration to run black and flake8.

### CODE REVIEW

File: trade_remedies_api/audit/utils.py [hunk 1]
```diff
@@ -78,6 +78,7 @@ def get_notify_fail_report(case=None, detail=False):
         audits = audits.filter(case_id=case.id)
     report = {}
     for audit in audits:
+        key = ""  # TODO this is added to stop error 'undefined name' Check what is the right fix
```
Comment (rossmiller) [inline trade_remedies_api/audit/utils.py:81]: Do we know the answer to this?

File: trade_remedies_api/cases/services/api.py [hunk 2]
```diff
@@ -455,14 +449,14 @@ def get(self, request, organisation_id=None, case_id=None, user_id=None, *args,
...
-    @transaction.atomic
+    @transaction.atomic  # noqa: C901
     def post(self, request, *args, **kwargs):
         product = None
         export_source = None
         case_id = request.data.get("id")
         request_params = request.data.dict()
         case_name = request.data.get("case_name")
-        # case_type_id = request.data.get('case_type_id')
+        case_type_id = request.data.get("case_type_id")
```
Comment (rossmiller) [inline trade_remedies_api/cases/services/api.py:459]: Why has this been uncommented?

File: trade_remedies_api/cases/services/api.py [hunk 3]
```diff
@@ -1469,7 +1473,7 @@ def post(self, request, organisation_id=None, case_id=None, *args, **kwargs):
         case_type_id = request.data.get("case_type")
         case = get_case(str(case_id))
         if case_type_id:
-            modified = case.modify_case_type(case_type_id, requested_by=request.user)
+            case.modify_case_type(case_type_id, requested_by=request.user)
```
Comment (rossmiller) [inline trade_remedies_api/cases/services/api.py:1476]: Why has this been changed?

File: trade_remedies_api/cases/services/urls.py [hunk 4]
```diff
@@ -147,10 +149,12 @@
         "<uuid:case_id>/organisation/<uuid:organisation_id>/exportsource/<uuid:export_source_id>/",
         ExportSourceAPIView.as_view(),
     ),
-    path("<uuid:case_id>/submission/<uuid:submission_id>/reviewtype/", ReviewTypeAPIView.as_view()),
+    path(
+        "<uuid:case_id>/submission/<uuid:submission_id>/reviewtype/", ReviewTypeAPIView.as_view()
+    ),  # noqa: E501
```
Comment (rossmiller) [inline trade_remedies_api/cases/services/urls.py:154]: This appears to be a mistake.

File: trade_remedies_api/contacts/services/api.py [hunk 5]
```diff
@@ -68,7 +68,7 @@ def post(self, request, contact_id=None, organisation_id=None, case_id=None, *ar
                 raise NotFoundApiExceptions("Contact does not exist or access is denied")
             if organisation_id:
                 organisation = Organisation.objects.get(id=organisation_id)
-                ccs = CaseContact.objects.get_or_create(
```
Comment (rossmiller) [inline trade_remedies_api/contacts/services/api.py:71]: Presumably this variable was not used?

File: trade_remedies_api/contacts/services/api.py [hunk 6]
```diff
@@ -97,7 +97,7 @@ def delete(self, request, contact_id=None, *args, **kwargs):
             )
         except Contact.DoesNotExist:
             raise NotFoundApiExceptions("Contact does not exist or access is denied")
-        result = CaseContact.objects.filter(contact=contact).delete()
```
Comment (rossmiller) [inline trade_remedies_api/contacts/services/api.py:100]: Presumably this variable was not used?

File: trade_remedies_api/core/management/commands/load_sysparams.py [hunk 7]
```diff
@@ -1,9 +1,9 @@
 import logging
 import json
 from core.models import SystemParameter
-from django.contrib.contenttypes.models import ContentType
 from django.core.management.base import BaseCommand
-from django.conf import settings
+
+logger = logging.getLogger(__name__)
```
Comment (rossmiller) [inline trade_remedies_api/core/management/commands/load_sysparams.py:6]: This looks like a mistake.

File: trade_remedies_api/core/services/auth.py [hunk 8]
```diff
@@ -250,7 +242,8 @@ def get(self, request, delivery_type=None, *args, **kwargs):
             return ResponseSuccess(
                 {
                     "result": {
-                        "error": "You have entered an incorrect code too many times and we have temporarily locked your account.",
+                        "error": "You have entered an incorrect code too many times and "
```
Comment (rossmiller) [inline trade_remedies_api/core/services/auth.py:245]: Does that structure function as expected?

File: trade_remedies_api/organisations/services/api.py [hunk 9]
```diff
@@ -220,15 +219,15 @@ class OrganisationContactsAPI(TradeRemediesApiView):
...
-    def get(self, request, organisation_id, case_id=None, *args, **kwargs):
+    def get(self, request, organisation_id, case_id=None, *args, **kwargs):  # noqa: C901
 
         contacts = {}
         case = Case.objects.get(id=case_id) if case_id else None
 
         def add_contact(case_contact, case):
             if not get(case_contact, "contact").deleted_at:
                 contact_id = str(get(case_contact, "contact").id)
-                if not contact_id in contacts:
```
Comment (rossmiller) [inline trade_remedies_api/organisations/services/api.py:231]: Does the replacement have the same logical effect?

File: trade_remedies_api/workflow/services/api.py [hunk 10]
```diff
@@ -24,14 +19,15 @@ def post(self, request, template_id=None, *args, **kwargs):
             template = WorkflowTemplate.objects.create(
                 name=request.data.get("template_name", "Untitled"), template=Workflow.BOILERPLATE
             )
+        # TODO flake8 fix F821 undefined name 'item'
```
Comment (rossmiller) [inline trade_remedies_api/workflow/services/api.py:22]: Is this TODO still required?

File: trade_remedies_api/workflow/models.py [hunk 11]
```diff
@@ -323,16 +327,17 @@ def workflow_state(self, workflow, **kwargs):
         elif isinstance(workflow, dict):
             workflow = Workflow(workflow)
         for key in workflow.key_index:
-            state.key_index[key]["value"] = self.current_value(key, **kwargs)
+            # TODO fix F821 (undefined name 'state') instead of using noqa
```
Comment (rossmiller) [inline trade_remedies_api/workflow/models.py:330]: Is this TODO still required?

**Review by rossmiller — CHANGES_REQUESTED** [review]
Very nice! Some comments to address.

File: trade_remedies_api/cases/services/api.py (hunk 3 above)
Comment (Luisella21) [reply trade_remedies_api/cases/services/api.py:1476]: modified was not used and was giving the flake8 errorF841 (local variable  assigned to but never used)

File: trade_remedies_api/audit/utils.py (hunk 1 above)
Comment (Luisella21) [reply trade_remedies_api/audit/utils.py:81]: I checked the code in more details, and the key is initialised correctly.

File: trade_remedies_api/cases/services/api.py (hunk 2 above)
Comment (Luisella21) [reply trade_remedies_api/cases/services/api.py:459]: Mistake!

File: trade_remedies_api/cases/services/urls.py (hunk 4 above)
Comment (Luisella21) [reply trade_remedies_api/cases/services/urls.py:154]: Fixed!

File: trade_remedies_api/contacts/services/api.py (hunk 5 above)
Comment (Luisella21) [reply trade_remedies_api/contacts/services/api.py:71]: Exactly.

File: trade_remedies_api/core/services/auth.py (hunk 8 above)
Comment (Luisella21) [reply trade_remedies_api/core/services/auth.py:245]: I restored the long line.

File: trade_remedies_api/workflow/services/api.py (hunk 10 above)
Comment (Luisella21) [reply trade_remedies_api/workflow/services/api.py:22]: No! Removed it.

File: trade_remedies_api/workflow/models.py (hunk 11 above)
Comment (Luisella21) [reply trade_remedies_api/workflow/models.py:330]: No, removed it.

File: trade_remedies_api/contacts/services/api.py (hunk 6 above)
Comment (Luisella21) [reply trade_remedies_api/contacts/services/api.py:100]: Correct.

File: trade_remedies_api/organisations/services/api.py (hunk 9 above)
Comment (Luisella21) [reply trade_remedies_api/organisations/services/api.py:231]: Yes, it does.

File: trade_remedies_api/core/management/commands/load_sysparams.py (hunk 7 above)
Comment (Luisella21) [reply trade_remedies_api/core/management/commands/load_sysparams.py:7]: It was giving the error 'import but not used'
I reinstated it, and used noqa to stop the flake8 error.

**Review by rossmiller — APPROVED** [review]
LGTM!

**Pull request merged**